channeldb/channel.go

```diff
@@ -343,6 +349,12 @@ type OpenChannel struct {
 	sync.RWMutex
 }
 
+// SetResolver sets resolveTCP to a resolver other than the default
+// net.ResolveTCPAddr resolver function.
+func SetResolver(resolver func(string, string) (*net.TCPAddr, error)) {
```

Comment:
The fact that something like this was required makes me think that we need to get rid of the usage of `net.ResolveTCPAddr` within this package all toghter. After-all, this is _database_ logic :p 

The only instance this is used is here: https://github.com/lightningnetwork/lnd/blob/master/channeldb/nodes.go#L278. One possibility would be to just read the raw bytes of the address, and add an accessor function which actually returns a `net.Addr`. This doesn't get rid of host resolution all together, but no longer requires a host lookup _each time a `channeldb. LinkNode` is read_.